### DISCUSSION TITLE
Can not use variable for key

### DISCUSSION THREAD

**Discussion by Mark-Krensel** [neuefische/questions#310] [Web] [answered]
Hello, I want to use a declared variable, which I get from a fetch request and insert it in a db pull command, as a key (array name). It doesn't seem to take any variable as key.

### declared variable
```jsx
const attribute = request.query.attribute;
```
### intended use
```jsx
 await Day.updateMany({}, { $pull: { attribute: { _id: dataPointId } } });
```

I already tried many combinations of brackets around ```attribute``` like:
```jsx
 await Day.updateMany({}, { $pull: { {attribute}: { _id: dataPointId } } });
await Day.updateMany({}, { $pull: { ${attribute}: { _id: dataPointId } } });
await Day.updateMany({}, { $pull: { (${attribute}): { _id: dataPointId } } });
await Day.updateMany({}, { $pull: { (attribute): { _id: dataPointId } } });
```

, but none of these work. Is it possible, that a key may not be a variable?

Regards, Mark

**Comment by ThomasF85** [comment, marked as answer]
Hi Mark,

please give square brackets a try:
```
await Day.updateMany({}, { $pull: { [attribute]: { _id: dataPointId } } });
```

**Reply by Mark-Krensel** [reply]
It works! --> WTF?!! 
Thank you!
